src/vs/workbench/common/viewlet.ts

```diff
@@ -5,4 +5,9 @@
 
 import {IComposite} from 'vs/workbench/common/composite';
 
-export interface IViewlet extends IComposite { }
+export interface IViewlet extends IComposite {
+	/**
+	 * Returns the minimal width needed to avoid any content horizontal truncation
+	 */
+	getOptimalWidth?(): number;
```

Comment:
I tried that and I get the following error:

```
*** Error: /home/maxime/github/vscode/src/vs/workbench/browser/parts/sidebar/sidebarPart.ts(81,10): Type 'IComposite' is not assignable to type 'IViewlet'.
  Property 'getOptimalWidth' is missing in type 'IComposite'.
```

which is related to the following code:

```typescript
	public getActiveViewlet(): IViewlet {
		return this.getActiveComposite();
	}
```

How should composites be transformed into viewlets?